src/cpp/rtps/transport/TCPTransportInterface.cpp

```diff
@@ -1842,6 +1873,22 @@ void TCPTransportInterface::fill_local_physical_port(
     }
 }
 
+void TCPTransportInterface::send_channel_pending_logical_ports(
+        const uint16_t& physical_port,
+        std::shared_ptr<TCPChannelResource>& channel)
+{
+    std::lock_guard<std::mutex> channelPendingLock(channel_pending_logical_ports_mutex_);
+    auto logical_ports = channel_pending_logical_ports_.find(physical_port);
+    if (logical_ports != channel_pending_logical_ports_.end())
+    {
+        for (auto logical_port : logical_ports->second)
+        {
+            channel->add_logical_port(logical_port, rtcp_message_manager_.get());
+        }
+        channel_pending_logical_ports_.erase(physical_port);
```

Comment:
There might be a big issue here if we call OpenOutputChannel with two locators that only differ in their address.

Consider the case of two remote participants in different machines with the same listening port.
Let's say we are in the LARGE_DATA mode, and the heuristic makes them both need to connect to the local participant.
Our local participant will call `OpenOutputChannel(AddrA, listen_port, logical_port1)`, and then `OpenOutputChannel(AddrB, listen_port, logical_port1)`.

When the connection from AddrA is made, logical_port1 will be added to the channel, and the physical port will be erased from the map.
When the connection from AddrB is made, the physical port is not in the map, so the logical port is not added to the second connection.

Reply:
I suppose this means that the key of the map should be the physical locator instead of the port

Reply:
Agree